src/App.tsx

```diff
@@ -1,93 +1,125 @@
+import { ErrorNotifications } from './components/ErrorNotifications';
+import { HeaderForm } from './components/HeaderForm';
+import { TodoFilter } from './components/TodosFilter';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+  const { user } = useContext(AuthContext);
+  const { textError, setTextError } = useContext(ContextTextError);
+  const { todos, setTodos, isAddingTodo } = useContext(ContextTodos);
+  const { setUser } = useContext(AuthContext);
 
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+  const userId = user?.id || 0;
+  const unCompletedTodos = useMemo(() => {
+    return [...todos].filter(todo => !todo.completed);
+  }, [todos]);
+  const completedTodos = useMemo(() => {
+    return [...todos].filter(todo => todo.completed);
+  }, [todos]);
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const loaderTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(userId);
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+      setTodos(todosFromServer);
+    } catch {
+      setTextError(ErrorType.GET);
+    } finally {
+      setIsLoadingTodos(false);
+    }
+  }, [userId]);
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const handlerClearLocalStorage = useCallback(() => {
+    localStorage.clear();
```

Comment:
for the future, it will be better to delete some special keys in local storage instead of deleting all data from local storage
application can have much more data in local storage

Reply:
I have left .clear() in this project, because I also use localeStorage for filter type. And this parameter will leave for other user, if I replace it for .removeItem().
But, thank you for advice))